src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -794,7 +800,9 @@ private bool AreMethodsEqual(MethodSymbol method, MethodSymbol other)
                 method = SubstituteTypeParameters(method);
                 other = SubstituteTypeParameters(other);
 
-                return _comparer.Equals(method.ReturnType, other.ReturnType) &&
+                // When deleting a method, the return type of method and other will be the same symbol
+                // which fails the SymbolComparer check (to avoid cycles) but are obviously still equal.
+                return (_comparer.Equals(method.ReturnType, other.ReturnType) || ReferenceEquals(method.ReturnType, other.ReturnType)) &&
```

Comment:
Any reason not to update the comparer's Equal with ReferenceEqual check?

Reply:
Other than my usual abundance of caution, no. Shall move